### PR TITLE
Make gitserver cleanup to use an internal actor to avoid security eve…

### PR DESCRIPTION
**Pull request by vrto**
### Problem

We've recently deployed a [change](https://github.com/sourcegraph/sourcegraph/pull/42653) that made "security events" a part of the audit log and enabled it by default on every environment. @mrnugget has noticed that the change caused a bunch of errors to appear in the logs during the same day. 

The issue has been [hotfixed](https://github.com/sourcegraph/sourcegraph/pull/42763) by rolling back [the bit of code](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/commit/58a59b188430c8ef92528b5411acf99928da6edb?visible=1) that caused the errors. Luckily, the errors haven't been in the [newly added audit-logging code](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/commit/040059239070cf54a956b720843897f50724f6be#diff-65ea563100c640f7eef39bb91d6285f9R118). Still, due to a unfortunate combination of circumstances, we've executed code that hasn't run in production for a long time and has caused the errors.

### Explanation

The errors occur when the [LogEventList](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=internal/database/security_event_logs.go&start_row=152&start_col=33&end_row=152&end_col=33&editor=JetBrains&version=v2.0.1) method receives one or more events that don't have a valid `UserID` _and_ don't have a valid `AnonymousUserID` either.

I've located one place in the codebase in the gitserver cleanup logic with the following execution path:

- [cleanupRepos](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=cmd/gitserver/server/cleanup.go&start_row=550&start_col=4&end_row=550&end_col=4&editor=JetBrains&version=v2.0.1) 
  - [setRepoSizes ](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=cmd/gitserver/server/cleanup.go&start_row=589&start_col=17&end_row=589&end_col=17&editor=JetBrains&version=v2.0.1)
    - [fetchRepos](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=cmd/gitserver/server/cleanup.go&start_row=626&start_col=17&end_row=626&end_col=17&editor=JetBrains&version=v2.0.1) 
      - [ListMinimalRepos](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=internal/database/repos.go&start_row=862&start_col=20&end_row=862&end_col=20&editor=JetBrains&version=v2.0.1) 
        - [StreamMinimalRepos](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=internal/database/repos.go&start_row=846&start_col=8&end_row=846&end_col=8&editor=JetBrains&version=v2.0.1) 
          - [logPrivateRepoAccessGranted](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=internal/database/repos.go&start_row=181&start_col=4&end_row=181&end_col=4&editor=JetBrains&version=v2.0.1) 
            - [boom](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=internal/database/security_event_logs.go&start_row=114&start_col=27&end_row=114&end_col=27&editor=JetBrains&version=v2.0.1) (the error is swallowed internally by SecurityEventLogs, but it gets logged)

### Solution

- use an internal actor in the gitserver cleanup process so [that this condition is met](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/security-events/gitserver-janitor-internal-actor&file=internal/database/repos.go&start_row=177&start_col=19&end_row=177&end_col=19&editor=JetBrains&version=v2.0.1)
- re-enable the security events in the audit log

## Test plan

- wrote a failing test that exposed the problem
- manually confirmed that I couldn't see any more errors locally

### CODE REVIEW

**Comment by vrto** [comment]
@mrnugget can you double-check that this works well for you locally? thanks!

**Review by thenamankumar — APPROVED** [review]

**Review by evict — APPROVED** [review]

**Review by sashaostrikov — APPROVED** [review]

File: cmd/gitserver/server/server.go [hunk 1]
```diff
@@ -510,7 +510,7 @@ func (s *Server) Handler() http.Handler {
 func (s *Server) Janitor(interval time.Duration) {
```
Comment (mrnugget) [inline cmd/gitserver/server/server.go:510]: You also need to pass the `ctx context.Context` in here. We already have a `ctx` in the `main()` function where this is called.

Comment (vrto) [reply cmd/gitserver/server/server.go:510]: @mrnugget and keep the internal actor creation within `Janitor`, or pass it all the way from main? I'd opt for the former.

Comment (mrnugget) [reply cmd/gitserver/server/server.go:510]: I'd change the callsite from this

https://github.com/sourcegraph/sourcegraph/blob/2ba0c4b6c665649584093b61d10838f8af091410/cmd/gitserver/main.go#L200

to this

```
	go gitserver.Janitor(context.WithInternalActor(ctx), janitorInterval)
```

**Pull request merged**